account_invoice_ubl/models/account_invoice.py

```diff
@@ -0,0 +1,328 @@
+                line_root, ns['cbc'] + 'InvoicedQuantity',
+                unitCode=uom_unece_code)
+        else:
+            quantity = etree.SubElement(
+                line_root, ns['cbc'] + 'InvoicedQuantity')
+        qty = iline.quantity
+        quantity.text = '%0.*f' % (qty_precision, qty)
+        line_amount = etree.SubElement(
+            line_root, ns['cbc'] + 'LineExtensionAmount',
+            currencyID=cur_name)
+        line_amount.text = '%0.*f' % (account_precision, iline.price_subtotal)
+        self._ubl_add_invoice_line_tax_total(
+            iline, line_root, ns, version=version)
+        self._ubl_add_item(
+            iline.name, iline.product_id, line_root, ns, type='sale',
+            version=version)
+        price_node = etree.SubElement(line_root, ns['cac'] + 'Price')
+        price_amount = etree.SubElement(
+            price_node, ns['cbc'] + 'PriceAmount', currencyID=cur_name)
+        price_unit = 0.0
+        # Use price_subtotal/qty to compute price_unit to be sure
+        # to get a *tax_excluded* price unit
+        if not float_is_zero(qty, precision_digits=qty_precision):
+            price_unit = float_round(
+                iline.price_subtotal / float(qty),
+                precision_digits=price_precision)
+        price_amount.text = '%0.*f' % (price_precision, price_unit)
+        if uom_unece_code:
+            base_qty = etree.SubElement(
+                price_node, ns['cbc'] + 'BaseQuantity',
+                unitCode=uom_unece_code)
+        else:
+            base_qty = etree.SubElement(price_node, ns['cbc'] + 'BaseQuantity')
+        base_qty.text = '%0.*f' % (qty_precision, qty)
+
+    def _ubl_add_invoice_line_tax_total(
+            self, iline, parent_node, ns, version='2.1'):
+        cur_name = self.currency_id.name
+        prec = self.currency_id.decimal_places
+        tax_total_node = etree.SubElement(parent_node, ns['cac'] + 'TaxTotal')
+        price = iline.price_unit * (1 - (iline.discount or 0.0) / 100.0)
+        res_taxes = iline.invoice_line_tax_ids.compute_all(
+            price, quantity=iline.quantity, product=iline.product_id,
+            partner=self.partner_id)
+        tax_total = float_round(
+            res_taxes['total_included'] - res_taxes['total_excluded'],
+            precision_digits=prec)
+        tax_amount_node = etree.SubElement(
+            tax_total_node, ns['cbc'] + 'TaxAmount', currencyID=cur_name)
+        tax_amount_node.text = '%0.*f' % (prec, tax_total)
+        if not float_is_zero(tax_total, precision_digits=prec):
+            for res_tax in res_taxes['taxes']:
+                tax = self.env['account.tax'].browse(res_tax['id'])
+                # we don't have the base amount in res_tax :-(
+                self._ubl_add_tax_subtotal(
+                    False, res_tax['amount'], tax, cur_name, tax_total_node,
+                    ns, version=version)
+
+    @api.multi
+    def get_delivery_partner(self):
```

Comment:
This method either. This is not used anymore in module. And if is an inheritance, class method is name def get_delivery_partner_id() in AccountInvoice class.